composition/composition.md

```diff
@@ -151,30 +159,77 @@ Now, I'm going to leave Component's instance method
 This is what Component's method returned: 473
 ```
 
+## One-to-One and One-to-Many Composition Relationships
+
+We just saw examples of one-to-one and one-to-many composition relationships.
+
+A one-to-one relationship is a relationship in which a composite class has an attribute representing a single instance of a component class as in this example:
+
+```python
+class ExampleComponent:
+
+    def __init__(self, name):
+        self.name = name
+
+class ExampleComposite:
+
+    def __init__(self, name, component):
+        self.name = name
+        self.component = component
+```
+
+A one-to-many relationship is a relationship in which a composite class has an attribute representing a _list_ of instances of a component class as in this example:
+
+```python
+class ExampleComposite:
+
+    def __init__(self, name, components):
+        self.name = name
+        self.components = components
+```
+
+Note that `components` is a list of `component` objects. As usual, we use good naming, here the plural `components`, to help us remember what we expect an attribute to contain.
+
 ## A Longer Example: Product and ShoppingCart
 
+Let's look at one more example of one-to-one and one-to-many composition relationships.
+
 ```python
 class Product:
-
+ 
     def __init__(self, price, quantity):
         self.price = price
         self.quantity = quantity
 
     def calculate_cost(self):
         return self.price * self.quantity
 
+class ShippingAddress:
+
+    def __init__(self, name, street_address, city, state, zip):
+        self.name = name
+        self.street_address = street_address
+        self.city = city
+        self.state = state
+        self.zip = zip
+
 class ShoppingCart:
 
-    def __init__(self, products):
+    def __init__(self, products, shipping_address):
         self.products = products
+        self.shipping_address = shipping_address
 
     def calculate_total_cost(self):
         total_price = 0.0
 
         for product in self.products:
+            #----Refer to this line for Check for Understanding Question 2----
```

Comment:
my only concern with this phrasing is that it depends on learn numbering. Personally, I prefer having the marker be something we explicitly control and reference from the question rather than depending on learn numbering the questions.